Tests/Form/FormMapperTest.php

```diff
@@ -429,6 +429,20 @@ public function testKeys()
         $this->assertSame(array('foo', 'baz'), $this->formMapper->keys());
     }
 
+    public function testFieldNameIsSanitized()
+    {
+        $this->contractor->expects($this->any())
+            ->method('getDefaultOptions')
+            ->will($this->returnValue(array()));
+
+        $this->formMapper
+            ->add('fo.o', 'bar')
+            ->add('ba__z', 'foobaz')
+        ;
+
+        $this->assertSame(array('fo__o', 'ba____z'), $this->formMapper->keys());
```

Comment:
@lalop The behavior was to get `fo__o` when `fo.o` is added but not `ba____z` when `ba__z` is added. This introduced a bc break and **MUST** be fix.